include/mesh/boundary_info.h

```diff
@@ -550,6 +557,20 @@ class BoundaryInfo : public ParallelObject
    */
   void build_shellface_boundary_ids(std::vector<boundary_id_type> & b_ids) const;
 
+  /**
+  * Update parent's boundary id list so that this information is consistent with
+  * its children
+  *
+  * This is useful when `_children_on_boundary = true`, and is used when the
+  * element is about to get coarsened i.e., in MeshRefinement::_coarsen_elements()
+  *
+  * Specifically, when we coarsen an element whose children have different boundary ids.
+  * In such scenarios, the parent will inherit the children's boundaries if at
+  * least two of them own a boundary while sharing the side of the parent.
+  * Otherwise, we delete the boundary from the children and the parent as well.
+  */
+  void transfer_boundary_ids_to_parent(const Elem * const elem);
```

Comment:
This is somewhat vague and confusing - I can't tell from the comment whether `elem` is expected to be the parent or each child, from the function name I'd expect the child, from the function implementation I'd expect the parent (except that non-parents are filtered out rather than being errors?).

Reply:
Should we change the name to `transfer_boundary_ids_from_children`, change the parameter name to `parent`?

Reply:
Great idea!